spec/controllers/api/v1/offers_controller_spec.rb

```diff
@@ -161,6 +161,20 @@
       get :show, id: offer.id
       expect(response.status).to eq(200)
     end
+
+    context 'polymorphic associations' do
+      let(:item) { create(:item, offer: offer) }
+      let!(:item_message) { create(:message, messageable: item) }
+      let!(:offer_message) { create(:message, messageable: offer) }
+      let!(:order_message) { create(:message, :with_order, sender: user) }
```

Comment:
Could you modify the `item`, `offer` to share the same ID ? That would allow this to test for the serializer bug I mentioned before :)